### PR TITLE
[google_maps_flutter_android] Fix Android lint warnings

### PR DESCRIPTION
**Pull request by mgmix**
Fixed the Android lint warning shown in `lint-baseline.xml` and removed the baseline.

Part of https://github.com/flutter/flutter/issues/88011

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by reidbaker** [comment]
Thank you for your contribution. It looks like the tests are failing. Before we review the PR, please see what you can do to resolve the test failures. If you are unsure how to proceed, please reach out for help on the #hackers-new channel.

@gmackall the tests are failing on the duplicate class issue I think you have seen before but I dont see any updated dependencies that would introduce that class of error.

**Comment by mgmix** [comment]
@reidbaker Thank you, I fixed the part that was causing the problem and it passed all the tests, could you please check it again?

**Comment by reidbaker** [comment]
2 small changes but this looks good. Once those are fixed I am ready to approve. 

Thank you for helping make flutter better :)

**Comment by gmackall** [comment]
> Thank you for your contribution. It looks like the tests are failing. Before we review the PR, please see what you can do to resolve the test failures. If you are unsure how to proceed, please reach out for help on the #hackers-new channel.
> 
> @gmackall the tests are failing on the duplicate class issue I think you have seen before but I dont see any updated dependencies that would introduce that class of error.

Sorry, missed this ping last week. Is the failure you are referring to the one with the message 

`Runtime JAR files in the classpath should have the same version. These files were found in the classpath:`

That [ran here?](https://github.com/flutter/packages/runs/12822754921) Asking because that's the most similar message I see but not sure if there was an identical one that failed, but the logs have since expired.

If so, I don't think that message is caused by this change, it's actually still there on the [latest successful test run](https://cirrus-ci.com/task/5793488306962432) if you scroll down to the "run lint" instruction.

File: packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java [hunk 1]
```diff
@@ -60,7 +60,7 @@
   private final int id;
   private final MethodChannel methodChannel;
   private final GoogleMapOptions options;
-  @Nullable private MapView mapView;
+  @Nullable MapView mapView;
```
[the remark this answers is not in the document]
Comment (mgmix) [reply packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: Thank you! I restored the private scope modification in the GoogleMapController.java file, and after replacing the anonymous class GoogleMap.OnMapLoadedCallback() with a lambda expression, I confirmed that the synthetic-accessor warning no longer occurs. [4cbbc06](https://github.com/flutter/packages/pull/3751/commits/4cbbc06cd24f6d7077158fc5485a388a37d3c9d8)

**Review by reidbaker — APPROVED** [review]

**Review by gmackall — APPROVED** [review]
This LGTM as well. Thanks for the contributions (here and in local auth)!

File: packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/TileProviderController.java [hunk 2]
```diff
@@ -18,9 +18,9 @@ class TileProviderController implements TileProvider {
 
   private static final String TAG = "TileProviderController";
 
-  private final String tileOverlayId;
-  private final MethodChannel methodChannel;
-  private final Handler handler = new Handler(Looper.getMainLooper());
+  final String tileOverlayId;
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/TileProviderController.java]: > Can you make these SyntheticAccessor violations protected?

@reidbaker Out of curiosity, why `protected`? `protected` is actually slightly more visible than the default Java access (package private).

**Pull request merged**